src/Symfony/Bundle/FrameworkBundle/Resources/config/mailer.php

```diff
@@ -78,6 +82,36 @@
         ->set('mailer.messenger_transport_listener', MessengerTransportListener::class)
             ->tag('kernel.event_subscriber')
 
+         ->set('mailer.dkim_signer', DkimSigner::class)
+            ->args([
+                abstract_arg('key'),
+                abstract_arg('domain'),
+                abstract_arg('select'),
+                abstract_arg('options'),
+                abstract_arg('passphrase'),
+            ])
+
+        ->set('mailer.smime_signer', SMimeSigner::class)
+            ->args([
+                abstract_arg('key'),
+                abstract_arg('certificate'),
+                abstract_arg('passphrase'),
+                abstract_arg('extraCertificates'),
+                abstract_arg('signOptions'),
+            ])
+
+        ->set('mailer.dkim_signer.listener', DkimSignedMessageListener::class)
+            ->args([
+                service('mailer.dkim_signer')->ignoreOnUninitialized(),
```

Comment:
why ignoreOnUninitialized?

Reply:
Hi Nicolas.
I did that because the signers are injected on the event listener despite they are not instantiated. I didn't find a way to instantiate the signers conditionally if the blocks `dkim_signer` or `smime_signer` are configured on the yaml or xml. TBH, I'm not very familiar with the `ContainerConfigurator`

Thanks for your feedback!

Reply:
what you should do it to defined the service here, without this condition
and in FrameworkExtension, if the signing is not enabled, you call container->removeDefinition

Reply:
`ignoreOnUnitialized` means that if the service is not already *instantiated* for another reason, the argument will be ignored. This does not make sense here (as nothing else uses the `mailer.dkim_signer` service to be able to instantiate it). It would not work at all right now.